pyogrio/raw.py

```diff
@@ -191,7 +206,93 @@ def read_arrow(
     dataset_kwargs = _preprocess_options_key_value(kwargs) if kwargs else {}
 
     try:
-        result = ogr_read_arrow(
+        with ogr_open_arrow(
```

Comment:
Could we use the `open_arrow` (instead of the lower level `ogr_open_arrow`)  here in `read_arrow` ? Then the generic boilerplate (checking for pyarrow being installed, processing the path, the dataset kwargs etc) only needs to be written once.